lib/lexer_mdx.mll

```diff
@@ -89,44 +65,35 @@ and cram_text section = parse
         newline lexbuf;
         `Section section :: cram_text (Some section) lexbuf }
   | "  " ([^'\n']* as first_line) eol
-      { let header = Some (Block.Header.Shell `Sh) in
+      { let start = Lexing.lexeme_start_p lexbuf in
+        newline lexbuf;
+        let header = "sh" in
         let requires_empty_line, contents = cram_block lexbuf in
         let contents = first_line :: contents in
-        let labels = [] in
-        let legacy_labels = false in
-        let loc = Location.curr lexbuf in
-        List.iter (fun _ -> newline lexbuf) contents;
+        let label_cmt = Some "" in
+        let legacy_labels = "" in
+        let end_ = Lexing.lexeme_start_p lexbuf in
         let rest = cram_text section lexbuf in
+        let loc = loc ~start ~end_ in
         let block =
-          match
-            Block.mk ~loc ~section ~header ~contents ~labels
+            Block.Raw.make ~loc ~section ~header ~contents ~label_cmt
               ~legacy_labels ~errors:[]
```

Comment:
I wonder if, when switching from `Block` to `Block.Raw` it wouldn't make sense to make `legacy_labels` into an `option`, instead of using `""` as a sentinel to represent emptiness.

Reply:
There might be a better representation but here I merely transmitted the raw data we're extracting in the lexer.

Also note that using options does not rule out `Some ""` so I think in this case, the option is a bit redundant.

Reply:
Also `""` is not a sentinel, it's the actual value we extract from the lexer when there are no legacy labels!

Reply:
Yes, I guess it depends on whether you consider `""` a valid legacy label or not, but fair enough - if this is what is extracted by the lexer it somewhat makes sense to pass this as-is to the parser to then be interpreted accordingly.